juju/client/facade.py

```diff
@@ -926,11 +926,22 @@ def generate_definitions(schemas):
     return definitions
 
 
-def generate_facades(schemas):
+def sortable_schema_version(version_string: str) -> Tuple[int, int, int]:
+    """Return a sorting key in the form (major, minor, micro) from a version string."""
+    # 'latest' is special cased in load_schemas and should come last
+    if version_string == 'latest':
+        return (9000, 9000, 9000)
+    # raise ValueError if string isn't in the format A.B.C
+    major, minor, micro = version_string.split('.')
+    # raise ValueError if major, minor, and micro aren't int strings
+    return (int(major), int(minor), int(micro))
+
+
+def generate_facades(schemas: Dict[str, List[Schema]]) -> Dict[str, Dict[int, codegen.Capture]]:
     captures = defaultdict(codegen.Capture)
 
     # Build the Facade classes
-    for juju_version in sorted(schemas.keys()):
+    for juju_version in sorted(schemas.keys(), key=sortable_schema_version):
```

Comment:
Is the idea here to go from oldest to newest because the latter iteration overwrites the ealier?

Reply:
Email reply isn't as cool as I was hoping. But yeah, I think it does overwrite.
The relevant line is:
```
 captures[schema.version].clear(cls_name)
```

With this PR I'm just slavishly trying to keep the original intent working. But maybe this being technically broken is an opportunity for us to make some bigger changes?

Certainly in terms of implementation it would be nice to get rid of all the inheriting from `dict`, and have those classes just do operations on an internal `dict` instead -- at most implement the specific `dict` methods that are expected to be used and forward them to the internal `dict` rather than inheriting every `dict` method and overriding some of them  ...

Reply:
Oh and note that `schema.version` is a facade version, like `7`.

`juju_version` is the Juju schema version, like `'3.1.10'`.